nvdaHelper/local/textUtils.cpp

```diff
@@ -75,3 +89,45 @@ bool calculateWordOffsets(wchar_t* text, int textLength, int offset, int* startO
 	return true;
 }
 
+bool calculateCharacterOffsets(wchar_t* text, int textLength, int offset, int* startOffset, int* endOffset) {

```

Comment:
This function contains a lot of duplicated code. I'd like to suggest the following:

1. Create one function: calculateUniscribeOffsets. It could take an extra parameter, e.g. `enum Unit { CHARACTER, WORD };`
2. Based on the new parameter, return character or word offsets, respectively.
3. Optionally, for backwards compat, you could keep calculateWordOffsets that calls the new function.